x-pack/test/cases_api_integration/security_and_spaces/tests/common/internal/user_actions_get_users.ts

```diff
@@ -0,0 +1,408 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { Cookie } from 'tough-cookie';
+import { User } from '@kbn/cases-plugin/common/api';
+import { UserProfile } from '@kbn/security-plugin/common';
+import { securitySolutionOnlyAllSpacesRole } from '../../../../common/lib/authentication/roles';
+import { getPostCaseRequest } from '../../../../common/lib/mock';
+import { createCase, deleteAllCaseItems, updateCase } from '../../../../common/lib/utils';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import { getCaseUsers } from '../../../../common/lib/user_actions';
+import { suggestUserProfiles, loginUsers } from '../../../../common/lib/user_profiles';
+import {
+  createUsersAndRoles,
+  deleteUsersAndRoles,
+  getUserInfo,
+} from '../../../../common/lib/authentication';
+import {
+  obsOnly,
+  obsOnlyRead,
+  obsSec,
+  obsSecRead,
+  secOnly,
+  secOnlyRead,
+  secOnlySpacesAll,
+  superUser,
+  noKibanaPrivileges,
+  globalRead,
+} from '../../../../common/lib/authentication/users';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const es = getService('es');
+
+  describe('user_actions_get_users', () => {
+    describe('no profiles', () => {
+      it('returns the users correctly without profile ids', async () => {
+        const postedCase = await createCase(supertest, getPostCaseRequest());
+        await changeCaseTitle({
+          supertest,
+          caseId: postedCase.id,
+          version: postedCase.version,
+          title: 'new title',
+        });
+
+        const { participants, users } = await getCaseUsers({ caseId: postedCase.id, supertest });
+
+        expect(participants).to.eql([{ username: 'elastic', full_name: null, email: null }]);
+        expect(users).to.eql([]);
+      });
+    });
+
+    describe('profiles', () => {
+      const secOnlyInfo: User = getUserInfo(secOnlySpacesAll);
+      let cookies: Cookie[];
+      let seUserProfile: UserProfile;
+      let superUserProfile: UserProfile;
+      let superUserHeaders: { Cookie: string };
+      let secOnlyHeaders: { Cookie: string };
+
+      before(async () => {
+        await createUsersAndRoles(
+          getService,
+          [secOnlySpacesAll],
+          [securitySolutionOnlyAllSpacesRole]
+        );
+      });
+
+      beforeEach(async () => {
+        cookies = await loginUsers({
+          supertest: supertestWithoutAuth,
+          users: [superUser, secOnlySpacesAll],
+        });
+
+        superUserHeaders = {
+          Cookie: cookies[0].cookieString(),
+        };
+
+        secOnlyHeaders = {
+          Cookie: cookies[1].cookieString(),
+        };
+
+        const suggestedSecUsers = await suggestUserProfiles({
```

Comment:
Because we do not get a profile by an identifier like the username or the profile ID. ES does not expose any API to do that. The only way to get a user profile based on the username is to do a search (aka `_suggest` API call). You can read more about the API here https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-suggest-user-profile.html

Reply:
> Because we do not get a profile by an identifier like the username or the profile ID. ES does not expose any API to do that.

understood, thanks! 👍